### ISSUE TITLE
PeerDisconnectTimeout

### ISSUE THREAD

**Issue by WikkidEdd** [Azure/azure-remote-rendering#72]
One of our users got a PeerDisconnectTimeout error yesterday and we've been digging through the logs and want to better understand what this means.

It appears that they had an internet connection drop out, this disconnected them from the multi-user session (photon networking). We respond to that by resetting everything including calling Disconnect() and ForgetSession on the active ARR session which responded with "Success". About 10 seconds later their internet connection was restored and they rejoined the multi-user session, opened the ARR session, and then called ConnectAsync which threw a RRException PeerDisconnectTimeout and a callback to RenderingSession.ConnectionStatusChanged with PeerDisconnectTimeout.

The documentation for PeerDisconnectTimeout says "Timed out while waiting for peer to close connection". So I'm wondering is the exception being thrown related to the prior call to Disconnect or is it relevant to the ConnectAsync call?

(This is what led us down the path of trying to improve our disconnect handling in #71 )

**Comment by FlorianBorn71** [comment]
Hi Edd, we will look into this more closely and let you know if we have some follow-up questions.
Cheers,
Florian

**Comment by ChristopherManthei** [comment]
Hi @WikkidEdd ,
The PeerDisconnectTimeout is the result of the Disconnect() call. As there was a network outage, the client could not gracefully disconnect. This error code should not have bleed into the next connect call, this is a bug on our side.
For now as a workaround, you can wait after calling Disconnect() until you get the ConnectionStatusChanged event.

Cheers,
Christopher

**Comment by ChristopherManthei** [comment]
Hi @WikkidEdd ,
we just released [ARR 1.0.60](https://github.com/Azure/azure-remote-rendering/releases/tag/1.0.60). With this the previous disconnect should no longer have any side effects on the next connect call.
We also changed the code to make sure ForgetSession / StopSession automatically disconnect from the active session before executing which should ensure that events are always observed in the same order.
Cheers,
Christopher
